### PR TITLE
Implement reduce_accumulators_per_key in BeamOperations.

### PR DESCRIPTION
**Pull request by sushkoy**
## Description
Implement reduce_accumulators_per_key in BeamOperations.

As the accumulators interface is still work in progress, this assumes the interface agreed in the documentation.

## Affected Dependencies
None

## How has this been tested?
- Added unit tests

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: pipeline_dp/pipeline_operations.py [hunk 1]
```diff
@@ -69,6 +69,20 @@ def sample_fixed_per_key(self, col, n: int, stage_name: str):
     def count_per_element(self, col, stage_name: str):
         pass
 
+    @abc.abstractmethod
+    def reduce_accumulators_per_key(self, col, stage_name: str):
+        """Reduce the input collection so that all elements per each key are merged.
```
Comment (dvadym) [inline pipeline_dp/pipeline_operations.py:74]: Nit: Reduce -> Reduces

File: pipeline_dp/pipeline_operations.py [hunk 2]
```diff
@@ -160,6 +174,19 @@ def sample_fixed_per_key(self, col, n: int, stage_name: str):
     def count_per_element(self, col, stage_name: str):
         return col | stage_name >> combiners.Count.PerElement()
 
+    def reduce_accumulators_per_key(self, col, stage_name: str = None):
+
```
Comment (dvadym) [inline pipeline_dp/pipeline_operations.py:178]: Please add 
TODO: use merge function from the accumulator framework.

File: tests/pipeline_operations_test.py [hunk 3]
```diff
@@ -75,6 +75,17 @@ def test_filter_by_key_pcollection_empty_public_keys(self, in_memory):
                                             "Public partition filtering")
             assert_that(result, equal_to(expected_result))
 
+    def test_reduce_accumulators_per_key(self):
+        with test_pipeline.TestPipeline() as p:
+            col = p | "Create PCollection" >> beam.Create([(1, 6, 1), (2, 7, 1),
```
Comment (dvadym) [inline tests/pipeline_operations_test.py:80]: Nit: splittig would make formatting easier
```
col = p | ....
col = self.ops.map_values(col, SumAccumulator, "Wrap into accumulators") # it simplifies a little bit
result = ....
```

**Review by dvadym — CHANGES_REQUESTED** [review]
Thanks, it looks good. I've left some minor comments

File: pipeline_dp/pipeline_operations.py (hunk 2 above)
Comment (sushkoy) [reply pipeline_dp/pipeline_operations.py]: Done.

File: tests/pipeline_operations_test.py (hunk 3 above)
Comment (sushkoy) [reply tests/pipeline_operations_test.py]: Good idea, done.

**Comment by sushkoy** [comment]
Resolved all the requested changes.

**Review by dvadym — APPROVED** [review]

**Comment by dvadym** [comment]
Thanks for implementing!

**Pull request merged**